game-core/src/main/java/games/strategy/debug/error/reporting/ErrorReportConfiguration.java

```diff
@@ -0,0 +1,36 @@
+package games.strategy.debug.error.reporting;
+
+import java.net.URI;
+import java.util.function.BiConsumer;
+
+import javax.swing.JFrame;
+
+import org.triplea.http.client.error.report.ErrorReportClientFactory;
+
+import games.strategy.engine.lobby.client.login.LobbyPropertyFetcherConfiguration;
+import games.strategy.triplea.settings.ClientSetting;
+
+class ErrorReportConfiguration {
+
+  /**
+   * Creates a 'report handler', this is the component invoked when we have collected all data from
+   * user and wish to then send that data. The report handler is responsible for showing a waiting
+   * dialog and sending error report data to the remote http server.
+   */
+  static BiConsumer<JFrame, UserErrorReport> newReportHandler() {
+    if (ClientSetting.httpLobbyUriOverride.isSet()) {
```

Comment:
Might be more idiomatic here to use `getValue()` and `map()` and `orElseGet()`:

```java
return ClientSetting.httpLobbyUriOverride.getValue()
    .<BiConsumer<JFrame, UserErrorReport>>map(
        httpLobbyUri -> new ErrorReportUploadAction(
            ErrorReportClientFactory.newErrorUploader(
                URI.create(httpLobbyUri))))
    .orElseGet(() -> LobbyPropertyFetcherConfiguration.lobbyServerPropertiesFetcher()
        .fetchLobbyServerProperties()
        .<BiConsumer<JFrame, UserErrorReport>>map(
            lobbyServerProperties -> new ErrorReportUploadAction(
                ErrorReportClientFactory.newErrorUploader(
                    lobbyServerProperties.getHttpServerUri())))
        .orElse(ErrorReportUploadAction.OFFLINE_STRATEGY));
```

Reply:
Thanks for poiting this out, I was able to run with the idea and added: 9df2a737e

The same chunk of code now looks like this:
```
  static BiConsumer<JFrame, UserErrorReport> newReportHandler() {
    return newFromUserOverride()
        .orElseGet(
            () -> newFromRemoteProperties().orElse(ErrorReportUploadAction.OFFLINE_STRATEGY));
  }

  private static Optional<BiConsumer<JFrame, UserErrorReport>> newFromUserOverride() {
    return ClientSetting.httpLobbyUriOverride
        .getValue()
        .map(URI::create)
        .map(ErrorReportClientFactory::newErrorUploader)
        .map(ErrorReportUploadAction::new);
  }

  private static Optional<BiConsumer<JFrame, UserErrorReport>> newFromRemoteProperties() {
    return LobbyPropertyFetcherConfiguration.lobbyServerPropertiesFetcher()
        .fetchLobbyServerProperties()
        .map(
            lobbyServerProperties ->
                new ErrorReportUploadAction(
                    ErrorReportClientFactory.newErrorUploader(
                        lobbyServerProperties.getHttpServerUri())));
  }
```

Reply:
:+1: